operator/pkg/parsed/module.go

```diff
@@ -88,5 +88,13 @@ func (m *Module) UpdateStatusFromCluster(ctx context.Context, clnt client.Client
 
 	m.Unstructured.Object["status"] = unstructuredFromServer.Object["status"]
 	m.Unstructured.SetResourceVersion(unstructuredFromServer.GetResourceVersion())
+	m.Unstructured.SetGeneration(unstructuredFromServer.GetGeneration())
```

Comment:
also here, to set the generation typically is a pattern that does not make sense since the generation will only change for spec changes. If you ever needed to set that I think there is a state management issue hidden somewhere

Reply:
removed